textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -124,10 +140,62 @@ public ITuple getNextTuple() throws DataFlowException {
                     }
 
                     boolean allTokenPresent = areAllTrue(tokensPresent);
+                    if(predicate.getOperatorType() == KeywordOperatorType.BASIC) {
+
+                        if (!allTokenPresent) {
+                            spanList.removeAll(spanForThisField);
+                        }
+                    }
+
+                    else if(predicate.getOperatorType() == KeywordOperatorType.PHRASE){
```

Comment:
Lucene can only provide positions of tokens in phrase query. The phrase query will keep changing thus phrase positioning cant be provided by lucene. We have to combine tokens.